book/specifications/kimchi/template.md

```diff
@@ -88,9 +88,17 @@ TODO: use expr to define the index columns?
 
 {sections.constraint_system}
 
-## Prover and Verifier Index Creation
+## Prover Index
 
-{sections.indexes}
+{sections.prover_index}
+
+## Verifier Index
+
+The verifier index is essentially a number of pre-computations containing:
+
+* the (non-hidding) commitments of all the required polynomials
```

Comment:
are these polynomials, partly, each of the gate constraints of the specific circuit? Meaning, the specification of the circuit, which is public information (thus non-hiding commitments being used).
Typo -> non-hiding, only 1 d I think

Reply:
Oh! so only those polynomials that will tell the verifier what gate appears in each row I see... The mutually exclusive polynomials you mean.

Well I have one question. I understand that the verifier checks that `quot(X) * van(X) = poly(x)`, where `poly` is the set of all constraints, permutations etc. Then, how does it check that `poly(X)` indeed corresponds to "the shape that it should"? Will it "rebuild" the equation checks using the committed witness together with the set of constraints? (but perhaps this approach takes linear time in the number of rows, right?) I don't know if I am making myself clear.

Reply:
that's what I call the polynomial dance between the prover and the verifier. The verifier doesn't want to just get poly (or f) from the prover, the verifier recreates an f from the constraints of each gate, the witness, the permutation polynomial, etc. Oh, and they do that hidden in the exponent, so as a commitment, not as a polynomial directly.